### ISSUE TITLE
More information from the output of `release-plz release -o json`

### ISSUE THREAD

**Issue by tgross35** [release-plz/release-plz#1865]
## Motivations

It would be helpful for other tooling to have a way of knowing which crates are ready for a release. This would make it easier to do something like build binaries only when `release-plz release` would publish a crate.

- Would you like to implement this feature? [y/n] n

## Solution

If `release-plz release -o json` would do a release, running the same command with `--dry-run` should produce the same output JSON:

```json
{
  "releases": [
    {
      "package_name": "<package_name>",
      "prs": "<prs>",
      "tag": "<tag_name>",
      "version": "<version>"
    }
  ]
}
```

## Alternatives

JSON output could be extended with other information that might be useful to other CLI tools:

```json5
{
  // Always present; lists packages managed by `release-plz`
  "packages": [
    {
      "package_name": "<package_name>"
    }
  ],

  // Existing key
  "releases": [ /* ... */ ],

  // Similar to `releases` except printed when a package fails
  "failed_releases": [
    {
      "package_name": "<package_name>",
      "prs": "<prs>",
      "tag": "<tag_name>",
      "version": "<version>",
      "reason": "..."
    }
  ]
}
```

## Additional context

https://github.com/release-plz/release-plz/issues/1857

**Comment by KekmaTime** [comment]
@marcoieni can i work on this issue

**Comment by marcoieni** [comment]
Sure!

**Comment by GirishRawat** [comment]
Hi! I'm a student in the CodePath AI301 program and I'd like to work on this issue. I'll start by reproducing the problem locally this week. Any pointers on where to start in the codebase would be appreciated!

**Comment by GirishRawat** [comment]
Hi @marcoieni does this issue still need help? I haven't really heard back from anyone for this. Can I claim it?

**Comment by marcoieni** [comment]
Yes. I don't have that much time, so I don't guarantee I review the pr
